README.md

```diff
@@ -0,0 +1,175 @@
+
+Then, link the python binary to Python 3.8 by running: `sudo ln -s -f /usr/local/bin/python3.8 /usr/local/bin/python`
+
+Make sure you have the right version of cantools. If you run into issues regarding this, you can try running the following commands
+
+```
+pip3 uninstall cantools
+pip3 install cantools==35.0.0
+```
+
+Make sure you are in the same directory as the requirements.txt file (firmware/common) and run the command: `pip3 install -r requirements.txt`
+
+If you are able to successfully run `make all` within the /firmware repository, you have been successful! Feel free to reach out on Slack if you need assistance.
+
+# Linux Set Up
+
+### Installation
+
+1. Install OpenOCD through [this link](https://sourceforge.net/projects/openocd/files/openocd/0.10.0/) or try the following command `sudo apt-get install openocd`
+2. If you want to use OpenOCD from Eclipse, follow the instruction under GNU/Linux (http://gnuarmeclipse.github.io/openocd/install/)
+3. Download the GNU ARM Embedded Toolchain (version:9-2019-q4-major) from [here](https://developer.arm.com/open-source/gnu-toolchain/gnu-rm/downloads)
+4. Once installed, follow the following commands to ensure you can access the toolchain
+
+```
+# Peep your current PATH
+echo $PATH
+# /usr/local/heroku/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:<BLA BLA BLA>
+# Change your PATH by appending the "bin" folder of the folder
+# NOTE: You should also add this line to ~/.bashrc or ~/.bash_profile (in your home directory)
+# so you don't have to remember to do this every time you want to compile firmware.
+export PATH="$PATH:/usr/local/gcc_arm/gcc-arm-none-eabi-7-2017-q4-major/bin/" # Check to ensure the gcc-arm firmw
+arm-none-eabi-gcc --version
+# arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 9-2019-q4-major) 9.2.1 20191025 (release) [ARM/arm-9-b
+# Copyright (C) 2019 Free Software Foundation, Inc.
+# This is free software; see the source for copying conditions. There is NO
+# warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+```
+
+5. Config your git with the following commands
+
+```
+git config --global core.autocrlf input
+git config --global user.email "your_email"
+git config --global user.name "your_name"
+```
+
+6. Clone the firmware repo using the following command: `git clone git@github.com:UWaterloo-Formula-Electric/firmware.git`
+7. Make sure you are in the same directory as the requirements.txt (firmware/common), and run `pip install -r requirements.txt`
+8. Run `make all` within the firmware repo to try and build the code :)
+
+# Other Resources
+
+1. Git tutorial: https://www.freecodecamp.org/news/what-is-git-learn-git-version-control/
+2. CAN Bus tutorial: https://www.kvaser.com/can-protocol-tutorial/
+3. Learning C, plenty of resources online
+4. STM32CubeMX (may be useful) installation: https://www.st.com/en/development-tools/stm32cubemx.html
```

Comment:
Would be really good to have a FAQ at the end here for the things that people bring up this term. Could even ask people to make a PR with their questions :D